src/Front/DesugarPatTup.hs

```diff
@@ -0,0 +1,35 @@
+module Front.DesugarPatTup where
+
+import qualified Common.Compiler as Compiler
+import Common.Identifiers
+import qualified Front.Ast as A
+import           Data.Bifunctor                 ( first )
+
+desugarTupPat :: A.Program -> Compiler.Pass A.Program
+desugarTupPat (A.Program decls) = return $ A.Program $ desugarTop <$> decls
+    where desugarTop (A.TopDef defs) = A.TopDef $ desugarDef defs
+          desugarTop t = t
+          desugarDef (A.DefFn i [] t e) = A.DefFn i [] t $ desugarExpr e
+          desugarDef (A.DefFn i ps t e) =
+                let (pats, rese) = desugarPatTup ps e (0::Int) in
+                    A.DefFn i pats t $ desugarExpr rese
+                where desugarPatTup [] ex _ = ([],ex)
+                      desugarPatTup (p:rps) ex n = 
```

Comment:
Nit: Try not to mix `where` and `let` clauses in the same expression; stick to one or the other.